### PR TITLE
Rename src/urllib3 to src/hip

### PR DESCRIPTION
**Pull request by pquentin**
This is the big rename pull request! After this one, we will have the docs left and various text files. Remaining tasks:

- [x] Use uppercase Hip when it makes sense
- [x] Fix docs build
- [x] Add src/hip/_sync/ in .gitignore (per #180)

### CODE REVIEW

**Comment by pquentin** [comment]
Oops, I did not realize that the docs was failing to build because I had urllib3 installed in my virtual environment.

**Comment by sethmlarson** [comment]
Do we want to change the casing of hip -> Hip for non-package / members / installable / module references?  urlli3b is deliberately all lowercase but Hip I don't is.

**Comment by pquentin** [comment]
You're right, I'll do this. :)

File: setup.cfg [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [metadata]
-name = hip
+name = Hip
```
Comment (sethmlarson) [inline setup.cfg]: I'd consider this field a package name field so would be "hip" (ie pip install hip)

Comment (pquentin) [reply setup.cfg]: Great catch, thanks!

**Comment by sethmlarson** [comment]
This is a solid chunk of work @pquentin!

**Comment by pquentin** [comment]
Thank you for the reviews, they were very helpful!

**Pull request merged**